pkg/operator/controller/controller.go

```diff
@@ -318,38 +351,45 @@ func (r *reconciler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress,
 		Controller: &trueVar,
 	}
 
-	lbService, err := r.ensureLoadBalancerService(ci, infraConfig, current)
-	if err != nil {
-		return fmt.Errorf("failed to ensure load balancer service for %s: %v", ci.Name, err)
-	}
-
-	err = r.ensureDNS(ci, dnsConfig, lbService)
+	errs := []error{}
+	lbService, err := r.ensureLoadBalancerService(ci, current)
 	if err != nil {
-		return fmt.Errorf("failed to ensure DNS for %s: %v", ci.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure load balancer service for %s: %v", ci.Name, err))
+	} else {
```

Comment:
`ensureDNS()` was designed to do the right thing with whatever the current LB state is. There's no semantic error analysis here. Just because an error was returned by `ensureLoadBalancerService` doesn't mean we can't also try ensuring DNS. It could be the LB was created before and now there's some transient error preventing a no-op reconcile.

Reply:
fixed